ykrt/src/compile/jitc_yk/aot_ir.rs

```diff
@@ -547,4 +591,72 @@ func bar;
         check("");
         check("         ");
     }
+
+    #[test]
+    fn const_int_strings() {
+        use num_traits::cast::NumCast;
+        use std::mem;
+
+        // Check (in an endian neutral manner) that a `num-bits`-sized integer of value `num`, when
+        // converted to a constant IR integer, then stringified, results in the string `expect`.
+        //
+        // When `num` has a bit size greater than `num_bits` the most significant bits of `num` are
+        // treated as undefined: they can be any value as IR stringification will ignore them.
+        fn check<T: NumCast + Sized>(num_bits: u32, num: T, expect: &str) {
+            dbg!(mem::size_of::<T>() * 8, usize::try_from(num_bits).unwrap());
+            assert!(mem::size_of::<T>() * 8 >= usize::try_from(num_bits).unwrap());
```

Comment:
`debug_assert` ?

Reply:
it's in a test, so i figured we may as well always check.

Reply:
Ah!